### ISSUE TITLE
Observed panics during cluster topology change

### ISSUE THREAD

**Issue by jstncbllr** [redis/rueidis#694]
While switching a cluster in AWS from redis to valkey, also modifying the topology, I observed a couple of panics like:

```
PANIC: runtime error: index out of range [423] with length 0

runtime/debug.Stack()
	/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.5.linux-amd64/src/runtime/debug/stack.go:24 +0x5e
golang.org/x/sync/singleflight.newPanicError({0x21c0a20, 0xc1983cad50})
	/go/pkg/mod/golang.org/x/sync@v0.9.0/singleflight/singleflight.go:44 +0x25
golang.org/x/sync/singleflight.(*Group).doCall.func2.1()
	/go/pkg/mod/golang.org/x/sync@v0.9.0/singleflight/singleflight.go:193 +0x34
panic({0x21c0a20?, 0xc1983cad50?})
	/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.5.linux-amd64/src/runtime/panic.go:770 +0x132
github.com/redis/rueidis.(*clusterClient).redirectOrNew(0xc002032000, {0xc20be37d4a, 0x51}, {0x2a415e0, 0xc0024a1e00}, 0x1a7, 0x1)
	/go/pkg/mod/github.com/redis/rueidis@v1.0.49/cluster.go:446 +0x4de
github.com/redis/rueidis.(*clusterClient).do(0xc002032000, {0x2a2d000, 0xc20a6deed0}, {0xc194081a40, 0x2000, 0x1a7})
	/go/pkg/mod/github.com/redis/rueidis@v1.0.49/cluster.go:478 +0x6b1
github.com/redis/rueidis.(*clusterClient).Do(0xc036798f78?, {0x2a2d000?, 0xc20a6deed0?}, {0xc194081a40?, 0x8f20?, 0x3679?})
	/go/pkg/mod/github.com/redis/rueidis@v1.0.49/cluster.go:461 +0x7b
```

Not sure why.  Perhaps the `SHARDS` command didn't return any groups for a short while?

Possibly there could be a more graceful behavior in the client in this case?

**Comment by rueian** [comment]
Hi @jstncbllr,

Thanks for reporting the issue. We are fixing it https://github.com/redis/rueidis/pull/695.

**Comment by rueian** [comment]
If you are in a hurry, you can try the fix with v1.0.52-alpha.2.
